### ISSUE TITLE
Add user defined identifying label to trusted-ca's

### ISSUE THREAD

**Issue by mbaeuerle** [eclipse/hono#2456]
The Hono tenant management API allows adding multiple trusted CAs which are then used as trust anchors to authenticate devices.
When adding a trust anchor one has to provide some information (amongst others):
* subject-dn
* not-before
* not-after

A tenant can however have multiple trusted CAs with the same subject-dn and different expiration dates.
Consider a tenant having the following CAs:
```json
[
  {
    "subject-dn":"CN=Doors,O=ACME Inc.",
    "not-before": "2018-01-01T00:00:00+01:00",
    "not-after": "2021-01-01T00:00:00+01:00"
  },
  {
    "subject-dn":"CN=Doors,O=ACME Inc.",
    "not-before": "2021-01-01T00:00:00+01:00",
    "not-after": "2025-01-01T00:00:00+01:00"
  }
]
```

Now consider the scenario where these trusted CAs are listed in a UI.
As they both have the same `subject-dn` it's very difficult to distinguish them. They could be listed as:
* CN=Doors,O=ACME Inc. (Validity: "2018-01-01T00:00:00+01:00"-"2021-01-01T00:00:00+01:00")
* CN=Doors,O=ACME Inc. (Validity: "2021-01-01T00:00:00+01:00"-"2025-01-01T00:00:00+01:00")

This however is very hard to read. 

A proposal therefore would be to introduce a _tenant unique identifier_ for each certificate which the user can define herself. Then this identifier or label could be listed in the UI and all the details in a sub-view (e.g. when clicking on it):
* doors-cert-valid-until-2021
* doors-cert-valid-until-2025

When adding a trusted CA the user would then also add the (mandatory) additional text field which has to be unique to the other CAs of this tenant (proposal: `cert-id`):
```
  {
    "cert-id": "doors-cert-valid-until-2025",
    "subject-dn":"CN=Doors,O=ACME Inc.",
    "not-before": "2021-01-01T00:00:00+01:00",
    "not-after": "2025-01-01T00:00:00+01:00"
  }
```

**Comment by sophokles73** [comment]
I do not see a problem with adding such behavior to the Registry Management API. However, I'd rather name the additional property *id* instead of *cert-id* because the trust anchor does not contain the certificate but its properties only.
I propose to define the property's value as unique among a tenant's trust anchors.
The property itself should be optional FMPOV. If no value is specified, the registry will consider the trust anchor as newly added and will assign a unique ID. WDYT?

**Comment by mbaeuerle** [comment]
These suggestions sound good to me!

**Comment by kaniyan** [comment]
> I propose to define the property's value as unique among a tenant's trust anchors.
> The property itself should be optional FMPOV. If no value is specified, the registry will consider the trust anchor as newly added and will assign a unique ID. WDYT?

Generating a unique ID if not provided sounds good. In that case, we can make use of the `randomUUID` as we do for the _device/tenant_ identifiers.

**Comment by kaniyan** [comment]
FMPOV it would be a breaking change to include this in a minor version release (1.7.0), if the above mentioned identifier is set to _unique_. Most probably a unique constraint should be imposed at the database level and the existing data should be migrated, so that this new field contains a unique value. In order to avoid that, we can introduce this new field without any unique constraint attribute in the minor release and in the next major release we could enforce that. @sophokles73 WDYT?

**Comment by kaniyan** [comment]
As we discussed, the addition of a new optional identifier doesn't introduce any breaking change in the Device Management API. So it can be included in a minor release. But the device registry implementations, might need to have a strategy to generate these ids for the existing data.

**Comment by kaniyan** [comment]
This has been closed via PR #2470.
